tests/utility/inf_sup_test.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utils.hpp>
+
+#include <cuco/detail/__config>
+#include <cuco/detail/utils.hpp>
+#include <cuco/extent.cuh>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <cstdint>
+#include <cuda/std/array>
+#include <limits>
+
+using T                        = uint64_t;
+__device__ auto constexpr data = cuda::std::array<T, 6>{1, 2, 3, 5, 6, 7};
+
+constexpr T none = std::numeric_limits<T>::max();  // denotes a missing value
+
+template <typename Extent>
+__global__ void infimum_kernel(T* result, Extent extent)
+{
+  auto const res = cuco::detail::infimum(data.begin(), data.end(), extent);
+
+  *result = (res != data.end()) ? *res : none;
+}
+
+template <size_t N>
+constexpr T compute_device_infimum(cuco::experimental::extent<T, N> extent)
+{
+  T res_h = none;
+  T* res_d;
+  CUCO_CUDA_TRY(cudaMalloc(&res_d, sizeof(T)));
+  infimum_kernel<<<1, 1>>>(res_d, extent);
+  CUCO_CUDA_TRY(cudaMemcpy(&res_h, res_d, sizeof(T), cudaMemcpyDeviceToHost));
+  CUCO_CUDA_TRY(cudaFree(res_d));
+
+  return res_h;
+}
+
+template <typename Extent>
+__global__ void supremum_kernel(T* result, Extent extent)
+{
+  auto const res = cuco::detail::supremum(data.begin(), data.end(), extent);
+
+  *result = (res != data.end()) ? *res : none;
+}
+
+template <size_t N>
+constexpr T compute_device_supremum(cuco::experimental::extent<T, N> extent)
+{
+  T res_h = none;
+  T* res_d;
+  CUCO_CUDA_TRY(cudaMalloc(&res_d, sizeof(T)));
+  supremum_kernel<<<1, 1>>>(res_d, extent);
+  CUCO_CUDA_TRY(cudaMemcpy(&res_h, res_d, sizeof(T), cudaMemcpyDeviceToHost));
+  CUCO_CUDA_TRY(cudaFree(res_d));
+
+  return res_h;
+}
```

Comment:
This is probably a bit oldschool. I might replace that with some thrust facilities to reduce the amount of boilerplate code.